modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts

```diff
@@ -2,65 +2,24 @@
  * Zero Knowledge Range Proofs as described in (Two-party generation of DSA signatures)[1].
  * [1]: https://reitermk.github.io/papers/2004/IJIS.pdf
  */
-import { createHash } from 'crypto';
-import BaseCurve from '../../curves';
+import { EcdsaRangeProof, BaseCurve } from '@bitgo/sdk-lib-mpc';
 import { PublicKey } from 'paillier-bigint';
-import { bitLength, randBits, randBetween } from 'bigint-crypto-utils';
-import { gcd, modInv, modPow } from 'bigint-mod-arith';
-import { DeserializedNtilde, DeserializedNtildeProof, RSAModulus, RangeProof, RangeProofWithCheck } from './types';
-import { bigIntFromBufferBE, bigIntToBufferBE } from '../../util';
-import { OpenSSL } from '../../../../openssl';
-
-// 128 as recommened by https://blog.verichains.io/p/vsa-2022-120-multichain-key-extraction.
-const ITERATIONS = 128;
+import { DeserializedNtilde, DeserializedNtildeProof, RangeProof, RangeProofWithCheck } from './types';
 
 /**
  * @deprecated Use generateSafePrimes from sdk-lib-mpc instead
  * @param bitLengths
  */
 export async function generateSafePrimes(bitLengths: number[]): Promise<bigint[]> {
-  const openSSL = new OpenSSL();
-  await openSSL.init();
-  const promises: Promise<bigint>[] = bitLengths.map((bitlength: number) => {
-    return openSSL.generateSafePrime(bitlength);
-  });
-  return await Promise.all(promises);
-}
-
-/**
- * @deprecated Use generateModulus from sdk-lib-mpc instead
- * @param bitlength
- */
-async function generateModulus(bitlength: number): Promise<RSAModulus> {
-  if (bitlength < 3072) {
-    // https://www.keylength.com/en/6/
-    console.warn('Generating a modulus with less than 3072 is not recommended!');
-  }
-  const bitlengthP = Math.floor(bitlength / 2);
-  const bitlengthQ = bitlength - bitlengthP;
-  const [p, q] = await generateSafePrimes([bitlengthP, bitlengthQ]);
-  const n = p * q;
-  // We never expect this to happen unless something went wrong with the wasm/openssl module
```

Comment:
I still think this comment is misleading at best (if it was copied into the new location for the generateModulus function). If we want to throw instead of just auto-retrying, I think that's fine, but you can't guarantee that multiplying two 1536 bit numbers together will result in a 3072 bit number. 

1111 * 1000 = 1111000 (7 bits)
1111 * 1111 = 11100001 (8 bits)

Reply:
You bring up a good point, I think we should retry if it doesnt hit 3072 instead of throwing. Our maximum needed bits will always be bitlengthP + bitlengthQ so we can keep retrying until the modulus hits that.

Reply:
Agreed, but lets make the retry count an optional input. We can add a default value if wed like.